### DISCUSSION TITLE
handleError method throws error: Object.assign requires that input parameter not be null or undefined

### DISCUSSION THREAD

**Discussion by alexmandrikdev** [alpinejs/alpine#3569] [1. Help]
Greetings!

According to our error tracker, the code below is causing an error in Safari browsers, stating that `Object.assign requires that input parameter not be null or undefined`. I'm afraid this issue hides many important error messages from us. I don't even know which part of the application throws the error that the `handleError` method wants to handle or which parameter is null or undefined.

https://github.com/alpinejs/alpine/blob/a0a6b74b7cd3f41d255176ade3f0bdfecdcaa666/packages/alpinejs/src/utils/error.js#L10

Can you help me fix this problem?

Thanks!

**Comment by SimoTod** [comment]
it should be `error` because the second parameter is always an object. Do you have a stack trace or are you just logging the exception message?

**Comment by alexmandrikdev** [comment]
The stack trace we have is relatively poor. I don't know why, but Flare doesn't use the sourcemaps uploaded by the sourcemap plugin.

https://flareapp.io/share/353bZL97#F1

**Comment by alexmandrikdev** [comment]
There is another error that may be related to this: https://flareapp.io/share/dPbqgRgm#F2

**Comment by joshhanley** [comment]
This issue has been converted to a discussion as we are now using discussions to track bugs and ideas.
See this announcement for more details https://github.com/alpinejs/alpine/discussions/1423

**Reply by alexmandrikdev** [reply]
Okay, thank you!

**Comment by ekwoka** [comment]
This is strange. it shouldn't ever have undefined or null unless someone is playing hard and fast with `throw`

In this case, this won't be doing much to reduce your logs, since the issue is that something is throwing nothing.

Like they just do `throw;` without actually throwing an error. or having the thing AFTER `throw` be on a new line (can't do that).

So its most likely in your own code, but could be another library underneath it. Not really supposed to throw nothing.

**Reply by alexmandrikdev** [reply]
The strange thing is I can't reproduce this error. We're just getting the error reports from Flare. I'm afraid it is reproducible only on specific devices/browsers.

**Reply by ekwoka** [reply]
It is quite strange.

I'll make a PR to handle bad throws.

**Reply by alexmandrikdev** [reply]
Thank you!
